seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/source/MongodbSourceReader.java

```diff
@@ -0,0 +1,103 @@
+import org.apache.seatunnel.connectors.seatunnel.common.source.AbstractSingleSplitReader;
+import org.apache.seatunnel.connectors.seatunnel.common.source.SingleSplitReaderContext;
+import org.apache.seatunnel.connectors.seatunnel.mongodb.config.MongodbParameters;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCursor;
+import org.bson.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class MongodbSourceReader extends AbstractSingleSplitReader<SeaTunnelRow> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MongodbSourceReader.class);
+
+    protected final SingleSplitReaderContext context;
+
+    private MongoClient client;
+
+    protected final MongodbParameters params;
+
+    protected final DeserializationSchema<SeaTunnelRow> deserializationSchema;
+
+    MongodbSourceReader(SingleSplitReaderContext context, MongodbParameters params, DeserializationSchema<SeaTunnelRow> deserializationSchema) {
+        this.context = context;
+        this.params = params;
+        this.deserializationSchema = deserializationSchema;
+    }
+
+    @Override
+    public void open() throws Exception {
+        client = MongoClients.create(params.getUri());
+    }
+
+    @Override
+    public void close() throws IOException {
+        if (client != null) {
+            client.close();
+        }
+    }
+
+    @Override
+    public void pollNext(Collector<SeaTunnelRow> output) throws Exception {
+        try (MongoCursor<Document> mongoCursor = client.getDatabase(params.getDatabase()).getCollection(params.getCollection()).find().iterator()) {
+
+            while (mongoCursor.hasNext()) {
+                Document doc = mongoCursor.next();
+                HashMap<String, Object> map = new HashMap<>(doc.size());
+                Set<Map.Entry<String, Object>> entries = doc.entrySet();
+                for (Map.Entry<String, Object> entry : entries) {
+                    if (!"_id".equalsIgnoreCase(entry.getKey())) {
+                        String key = entry.getKey();
+                        Object value = entry.getValue();
+                        map.put(key, value);
+                    }
+                }
+                String content = JsonUtils.toJsonString(map);
+                if (deserializationSchema != null) {
+                    deserializationSchema.deserialize(content.getBytes(), output);
+                } else {
+                    // TODO: use seatunnel-text-format
+                    output.collect(new SeaTunnelRow(new Object[]{content}));
+                }
+            }
+        } finally {
+            if (Boundedness.BOUNDED.equals(context.getBoundedness())) {
+                // signal to the source that we have reached the end of the data.
+                LOGGER.info("Closed the bounded http source");
```

Comment:
Mongo source.

Reply:
@TyrantLucifer Thank you for your reminder